File: 02_selection_sort/selection_sort.py
# Find the smallest elemento in the array
def findSmallest(arr):
    # Stores the smallest element
    smallest = arr[0]
    # Stores the index of the smallest element
    smallest_index = 0

    for i in range(1,len(arr)):
        if(arr[i] < smallest):
            smallest_index = i
            smallest = arr[i]
    
    return smallest_index

# Sort arrays
def selectionSort(arr):
    newArray = []
    for i in range(len(arr)):
        # Find the smallest element and stores it to the new array
        smallest = findSmallest(arr)
        newArray.append(arr.pop(smallest))
    return newArray


print(selectionSort([10,5,2,8,4,3]))

File: 02_selection_sort/selection_sort.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int findSmallest(int *arr, int size);
int *selectionSort(int *arr, int size);

int main(){
    int array[10] = {10,6,28,9,4,3,2,1,0,7};
    int size = sizeof(array)/sizeof(array[0]);
    int *sorted = selectionSort(array, size); 
    for(int i = 0; i < size; i++){
        printf("\n%d", sorted[i]);
    }
    return 0;
}

// Find the smallest value in the array
int findSmallest(int *arr, int size){
    // Store the smallest value
    int smallest = arr[0];
    // Store the index of the smallest value 
    int smallest_index = 0;
    for(int i = 1; i < size; i++){
        if(arr[i] < smallest){
            smallest = arr[i];
            smallest_index = i;
        }
    }
    return smallest_index;
}

int *selectionSort(int *arr, int size){
    // Creating a new array
    int *newarr = (int *)malloc(size*sizeof(int));
    for(int i = 0; i < size; i++){
        int smallest = findSmallest(arr, size);
        newarr[i] = arr[smallest];
        // Specifying the maximum size of the item
        arr[smallest] = INT_MAX;
    }
    return newarr;
}
